cygwin buildfix

baentsch opened the pull request:
Implements the cygwin-specific build remediation documented in https://github.com/openssl/openssl/issues/19531. Without this, "master" fails to build on cygwin.

t8m commented:
Could you please amend the commit message to add `Fixes #19531` ?

baentsch commented:
> Could you please amend the commit message to add `Fixes #19531` ?

Done.

baentsch commented:
Thanks for the explanation. Will do my best to remember for next PR commit. Or rather, may it be an idea to update https://github.com/openssl/openssl/blob/master/.github/PULL_REQUEST_TEMPLATE.md with more definite reminders for all contributor's benefit (what to check & do for a PR)?

I also understand that I won't have to do anything on this PR any more: Please correct me if I'm wrong.

baentsch commented:
Just found this problem when doing an `oqsprovider` CI job for `cygwin`... Would a PR be welcome adding github-action-based `cygwin` testing to this project as well? Or even add it to this PR?

t8m commented:
Adding it to this PR would be OK.

tmshort commented on Configurations/10-main.conf:
```diff
@@ -1717,6 +1717,7 @@ my %targets = (
         CFLAGS           => picker(default => "-Wall",
                                    debug   => "-g -O0",
                                    release => "-O3"),
+        ex_libs          => add("-lcrypt32"),
```
Noting that the 64bit version of this library is still called `crypt32.lib`, but in a different architecture directory.

baentsch replied:
Goodness, this. Accordingly we should successfully be able to test "win32" and "win64". See in https://github.com/openssl/openssl/commit/f09f774fbb35f03866f0e463a815032c1b7d7375 .

tmshort replied:
It was really just a note to me, to not confuse 32-bit vs 64-bit... sigh.

tmshort commented on providers/implementations/storemgmt/winstore_store.c:
```diff
@@ -23,7 +23,9 @@
 #include "file_store_local.h"
-
+#ifdef __CYGWIN__
+#include <Windows.h>
```
Nit: should this be capitalized?

baentsch replied:
Just to check capitalization doesn't matter on Windows :-) Fund aside: Changed in https://github.com/openssl/openssl/commit/f09f774fbb35f03866f0e463a815032c1b7d7375. Will be disappointed if it fails...

tmshort reviewed (approved):
Just a nit.

baentsch commented:
> Adding it to this PR would be OK.

So done in https://github.com/openssl/openssl/commit/f09f774fbb35f03866f0e463a815032c1b7d7375.

baentsch commented:
CI now passes (and fails without the code changes), so asking for a new review @t8m @tmshort

tmshort commented:
The new CI is failing...

baentsch commented:
> The new CI is failing... Issues with `isspace()` that weren't fixed in #21127

Just saw that: Would need to get me a cygwin machine to go through all these issues. Alternatively disable strict checking in this PR again: Is it an issue/goal of relevance to the OpenSSL team to.pass 'strict' build on cygwin?

baentsch commented:
> The new CI is failing... Issues with `isspace()` that weren't fixed in #21127

Can re-enable strict checks here after #21151 has landed.

t8m commented:
Instead of adding another .yml for the test, could you add the cygwin job to the Windows Github CI .yml?

baentsch commented:
> Instead of adding another .yml for the test, could you add the cygwin job to the Windows Github CI .yml?

Sure. Will do after #21151 has landed to test all together. Please let me know whether you're OK that we're not running "make test" as part of this CI: quic, symbols and simpletest (timer) fail under cygwin. Otherwise, I'll need to create a few separate issues as all those failures puzzle me (and I don't know QUIC, the OSSL-specific use of `nm` and the timer precision mechanisms under `cygwin` sufficiently well, respectively).

baentsch commented:
> The new CI is failing... Issues with `isspace()` that weren't fixed in #21127

@tmshort: Now fixed. strict checks now passing.

> Instead of adding another .yml for the test, could you add the cygwin job to the Windows Github CI .yml?

@t8m: Now added to `windows.yml`. `cygwin` tests passing.

Please let me know if you'd still expect me to do anything on this PR.

t8m commented on providers/implementations/storemgmt/winstore_store.c:
```diff
@@ -23,7 +23,9 @@
 #include "file_store_local.h"
-
+#ifdef __CYGWIN__
+#include <windows.h>
```
Please add one space between `#` and include

baentsch replied:
done in https://github.com/openssl/openssl/pull/21006/commits/aa85bd19afa3445d7dbb947564ebecb2148e8fc4

t8m reviewed (approved):
Approved, we will squash the commits when merging.

tmshort reviewed (approved):
Looks fine, but there was still a failing test, seemingly not related to these changes? I'll approve, pending the CI.

EDIT: It looks as though `check-ansi` is failing in a number of other PRs.

openssl-machine commented:
This pull request is ready to merge

t8m commented:
Squashed and merged to master branch. Thank you for your contribution.
